src/redist/targets/GenerateMSIs.targets

```diff
@@ -0,0 +1,91 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="14.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <Target Name="SetDotnetCoreUninstallBrandingInfo">
+    <PropertyGroup>
+      <BuildName>SDK.Uninstall.Tool</BuildName>
+      <DotnetCoreUninstallBrandName>Microsoft .NET Core SDK Uninstall Tool</DotnetCoreUninstallBrandName>
+    </PropertyGroup>
+  </Target>
+
+  <Target Name="SetupWixProperties" DependsOnTargets="SetupFileExtensions;GetCurrentRuntimeInformation">
+    <!-- AcquireWix Properties -->
+    <PropertyGroup>
+      <WixVersion>3.10.4</WixVersion>
+      <WixDownloadUrl>https://dotnetcli.azureedge.net/build/wix/wix.$(WixVersion).zip</WixDownloadUrl>
+      <WixRoot>$(ArtifactsDir)Tools/WixTools/$(WixVersion)</WixRoot>
+      <WixDestinationPath>$(WixRoot)/WixTools.$(WixVersion).zip</WixDestinationPath>
+      <WixDownloadSentinel>$(WixRoot)/WixDownload.$(WixVersion).sentinel</WixDownloadSentinel>
+    </PropertyGroup>
+
+    <!-- Generate MSI/Bundle Properties -->
+    <PropertyGroup>
+      <DotnetCoreUninstallGenerateMsiPowershellScript>$(MSBuildThisFileDirectory)packaging/windows/dotnetCoreUninstall/generatemsi.ps1</DotnetCoreUninstallGenerateMsiPowershellScript>
+    </PropertyGroup>
+
+    <PropertyGroup>
+      <ArtifactNameWithVersionDotnetCoreUninstall>dotnet-core-uninstall-$(Version)</ArtifactNameWithVersionDotnetCoreUninstall>
+      <DotnetCoreUninstallMSIInstallerFile>$(ArtifactsShippingPackagesDir)$(ArtifactNameWithVersionDotnetCoreUninstall)$(InstallerExtension)</DotnetCoreUninstallMSIInstallerFile>
+    </PropertyGroup>
+  </Target>
+
+  <Target Name="MsiTargetsSetupInputOutputs"
+          DependsOnTargets="GenerateLayout;SetupWixProperties;GetDotnetCoreUninstallGitCommitInfo">
+
+    <GenerateMsiVersion CommitCount="$(GitCommitCount)"
+                        VersionMajor="$(VersionMajor)"
+                        VersionMinor="$(VersionMinor)"
+                        VersionPatch="$(VersionPatch)">
+      <Output TaskParameter="MsiVersion" PropertyName="MsiVersion" />
+    </GenerateMsiVersion>
+
+    <GenerateGuidFromName Name="$(DotnetCoreUninstallMSIInstallerFile)">
```

Comment:
What you a saying here with the upgradecode is that each MSI is a unique product - no upgrade in-place. That necessary means the installation paths need to be different or if not different, the version of the file(s) needs to be greater than the one before it.

Reply:
hm... that's not what i want. It should just the bigger version can upgrade the older version. So I should just put the same guid here all the time?

Reply:
What is the installation behavior you want? I think this will work as long as the file versions keep going up - an example of this is the SharedHost MSI.

Reply:
I just put a hard coded value. Is should be good?